server/embed/etcd.go

```diff
@@ -307,6 +309,12 @@ func print(lg *zap.Logger, ec Config, sc config.ServerConfig, memberInitialized
 	if quota == 0 {
 		quota = storage.DefaultQuotaBytes
 	}
+	var autoCompactionRetention zapcore.Field
+	if sc.AutoCompactionMode == compactor.ModePeriodic {
```

Comment:
There is type alias for `compactor.ModePeriodc` in server/embed: `CompactorModePeriodic`.

Please consider to use `switch` branch because that mode can be invalid, like empty.